src/silo/preprocessing/preprocessor.cpp

```diff
@@ -0,0 +1,515 @@
+            SPDLOG_ERROR(return_code->GetError());
+            throw PreprocessingException(return_code->GetError());
+         }
+      }
+   } else {
+      std::string order_by_select = ", partitioned_metadata." + database_config.schema.primary_key +
+                                    " as " + database_config.schema.primary_key;
+      if (database_config.schema.date_to_sort_by.has_value()) {
+         order_by_select += ", partitioned_metadata." +
+                            database_config.schema.date_to_sort_by.value() + " as " +
+                            database_config.schema.date_to_sort_by.value();
+      }
+
+      for (const auto& [seq_name, reference_sequence] :
+           reference_genomes.raw_nucleotide_sequences) {
+         preprocessing_db.generateNucSequenceTable(
+            "raw_nuc_" + seq_name,
+            reference_sequence,
+            preprocessing_config.getNucFilenameNoExtension(seq_name).replace_extension(
+               silo::preprocessing::FASTA_EXTENSION
+            )
+         );
+
+         auto return_code = preprocessing_db.query(fmt::format(
+            R"-(
+create or replace view nuc_{0} as
+select key, sequence,
+partitioned_metadata.partition_id as partition_id
+{2}
+from raw_nuc_{0} raw, partitioned_metadata
+where raw.key = partitioned_metadata.{1};)-",
+            seq_name,
+            database_config.schema.primary_key,
+            order_by_select
+         ));
+
+         if (return_code->HasError()) {
+            SPDLOG_ERROR(return_code->GetError());
+            throw PreprocessingException(return_code->GetError());
+         }
+      }
+
+      for (const auto& [seq_name, reference_sequence] : reference_genomes.raw_aa_sequences) {
+         preprocessing_db.generateNucSequenceTable(
+            "raw_gene_" + seq_name,
+            reference_sequence,
+            preprocessing_config.getGeneFilenameNoExtension(seq_name).replace_extension(
+               silo::preprocessing::FASTA_EXTENSION
+            )
+         );
+         silo::FastaReader fasta_reader(preprocessing_config.getGeneFilenameNoExtension(seq_name)
+                                           .replace_extension(silo::preprocessing::FASTA_EXTENSION)
+         );
+
+         auto return_code = preprocessing_db.query(fmt::format(
+
+            R"-(
+create or replace view gene_{0} as
+select key, sequence,
+partitioned_metadata.partition_id as partition_id
+{2}
+from raw_gene_{0} raw, partitioned_metadata
+where raw.key = partitioned_metadata.{1};
+)-",
+            seq_name,
+            database_config.schema.primary_key,
+            order_by_select
+         ));
+      }
+   }
+
+   preprocessing::Partitions partition_descriptor = preprocessing_db.getPartitionDescriptor();
+
+   SPDLOG_INFO("preprocessing - building database");
+
+   return buildDatabase(partition_descriptor, reference_genomes, order_by_clause, alias_key);
+}
+
+void Preprocessor::buildPartitioningTable() {
```

Comment:
How about splitting this into two functions?
```
void Preprocessor::buildPartitioningTable() {
   if (database_config.schema.partition_by.has_value()) {
      return buildPartitioningTableByColumn(database_config.schema.partition_by.value())
   }
   
   buildEmptyPartitioning()
}
```
(naming can probably be improved)